### PR TITLE
calc_pvt called with 0 sats on startup

### PR DESCRIPTION
**Pull request by ungureanuvladvictor**
Fixes https://github.com/swift-nav/piksi_firmware/issues/238

/cc @fnoble

### CODE REVIEW

File: src/solution.c [hunk 1]
```diff
@@ -359,6 +410,10 @@ static msg_t solution_thread(void *arg)
       memcpy(nav_meas_old, nav_meas, sizeof(nav_meas));
       n_ready_old = n_ready;
 
+      if (n_ready_tdcp == 0) {
+        return 0;
```
Comment (fnoble) [inline src/solution.c]: hmm, should this actually be `continue`?

Comment (fnoble) [inline src/solution.c]: can also change the `     if (n_ready >= 4) {` check to do the same thing.

File: src/solution.c [hunk 2]
```diff
@@ -314,6 +314,50 @@ static void timer_set_period_check(uint32_t timer_peripheral, uint32_t period)
   __asm__("CPSIE i;");
 }
 
+static void simulation_thread()
```
Comment (fnoble) [inline src/solution.c]: maybe pick a different name, e.g. `solution_simulation` or something. `_thread` to me implies this is its own thread but really its running from the solution thread.

**Comment by fnoble** [comment]
How about making this line an early continue too:

https://github.com/swift-nav/piksi_firmware/pull/466/files#diff-a8b8debb0aca96bb1c1d74a97c2afaffR391

Also, what testing have you done on this?

**Comment by ungureanuvladvictor** [comment]
Just bench testing. Simulation mode works. Needs HITL now.

**Comment by fnoble** [comment]
HITL looks good!

**Pull request merged**